api/app/serializers/spree/v2/storefront/product_serializer.rb

```diff
@@ -9,15 +9,15 @@ class ProductSerializer < BaseSerializer
                    :updated_at
 
         attribute :purchasable do |product|
-          product.purchasable?
+          product.variants_including_master.any? { |variant| variant.purchasable? }
         end
 
         attribute :in_stock do |product|
-          product.in_stock?
+          product.variants_including_master.any? { |variant| variant.in_stock? }
         end
 
         attribute :backorderable do |product|
-          product.backorderable?
+          product.variants_including_master.any? { |variant| variant.backorderable? }
```

Comment:
Uhm, why would we move it out of the model?

Reply:
is this part of business logic?

Reply:
Why not?
It kinda makes sense to know whether product is backorderable or in stock.
Ideally ActiveRecord classes should handle persistence, associations and not much else but we can't really refactor them at this point of time.